src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -71,11 +72,49 @@ class PetDaoTest @Autowired constructor(private val sql: DSLContext) {
         assertEquals(newOwnerId, petDao.getAllPetsByCompanyId(companyId)[0].ownerId)
     }
 
+    //sql 2 tests
+    @Test
+    fun `get pets by owner`() {
+        val ownerId = 2L
+        val petCreationRequest1 = PetCreationRequest("Tom", PetType.CAT, companyId, LocalDate.now(), ownerId)
+        val petCreationRequest2 = PetCreationRequest("Luke", PetType.DOG, companyId, LocalDate.now(), ownerId)
+        petDao.createNewPet(petCreationRequest1)
+        petDao.createNewPet(petCreationRequest2)
+        val expectedResult = listOf(petCreationRequest1, petCreationRequest2).map { it.name }
+        assertEquals(expectedResult, petDao.getPetsByOwner(ownerId, companyId).map { it.name })
+    }
+
+    @Test
+    fun `get pets by owner when owner id are different`() {
+        val ownerId = 2L
+        petDao.createNewPet(pet1)
+        petDao.createNewPet(pet2)
+        val expectedResult = listOf(pet2).map { it.name }
```

Comment:
why only pet2? why is it different?

Reply:
The difference is the ownerId, pet1 has ownerId = 1 and pet2 has ownerId = 2.
When I call getPetsByOwner Im calling with ownerId = 2